actions/connect_test.go

```diff
@@ -92,15 +99,70 @@ func TestCommunicate(t *testing.T) {
 			sent: []*api.AgentReply{},
 		}
 
+		go cc.communicate(ctx, &bs)
+
+		time.Sleep(10 * time.Millisecond)
+		cancel()
+
+		assert.Equal(t, &api.AgentReply{
+			JobId:    5,
+			Sequence: api.Sequenece_EXECUTE,
+			Type:     api.ReplyType_SUCCESS,
+		}, bs.sent[0])
+	})
+
+	t.Run("Test execute test plugin fails", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		bs := blockingStream{
+			msg: &api.Action{
+				JobId:    5,
+				Action:   "fail",
+				Sequence: api.Sequenece_EXECUTE,
+			},
+			sent: []*api.AgentReply{},
+		}
+
+		go cc.communicate(ctx, &bs)
+
+		time.Sleep(10 * time.Millisecond)
+		cancel()
+
+		assert.Equal(t, &api.AgentReply{
+			JobId:    5,
+			Sequence: api.Sequenece_EXECUTE,
+			Type:     api.ReplyType_ERROR,
+			Message:  "Could not execute",
+		}, bs.sent[0])
+	})
+	t.Run("Test execute plugin in dry-run mode", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		bs := blockingStream{
+			msg: &api.Action{
+				JobId:    5,
+				Action:   "test",
+				Sequence: api.Sequenece_EXECUTE,
+			},
+			sent: []*api.AgentReply{},
+		}
+		c := Connector{
```

Comment:
For consistency reasons `c` should be `cc` here as well?

Reply:
this one is "local" thats why I changed it